src/ashley/validators.py

```diff
@@ -39,3 +41,20 @@ def error(self, message):
     def handle_data(self, data):
         """Concatenate the plain text found in each element."""
         self.text += data
+
+
+def validate_upload_image_file_size(file):
+    """Controls the size of the uploaded file is not over the limit authorized"""
+    if file.size > settings.MAX_UPLOAD_FILE_MB * 1024 * 1024:
+        raise ValidationError(
+            f"The maximum file size that can be uploaded is {settings.MAX_UPLOAD_FILE_MB}MB"
+        )
+    return file
+
+
+def validate_upload_image_allowed_type(file):
+    """Gets the extension of the file and check if it is authorized"""
+    ext = (os.path.splitext(file.name)[1])[1:]
+    if f"image/{ext}" not in settings.IMAGE_TYPE_ALLOWED:
```

Comment:
The setting `IMAGE_TYPE_ALLOWED` is a list of authorized MIME types, which is great.

But it is used here as a file extension checker.
There is not always an exact match between a mime type and the associated file extension.